### PR TITLE
UCP/CORE: add memh logging

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
add memh logging

## Why ?
for development and debug purposes

### CODE REVIEW

File: src/ucp/core/ucp_mm.c [hunk 1]
```diff
@@ -431,6 +431,7 @@ ucp_memh_get_slow(ucp_context_h context, void *address, size_t length,
                                        uct_flags);
     if (status == UCS_OK) {
         *memh_p = memh;
+        ucs_trace("memh %p: get slow", *memh_p);
```
Comment (yosefe) [inline src/ucp/core/ucp_mm.c:434]: 1. maybe add details such as address, length, md_map?
2. swap with line 433
3. maybe smth like  `ucs_trace("memh %p: registered address %p length %zu md_map %"PRIx64, ..., *memh_p);`

File: src/ucp/core/ucp_mm.inl [hunk 2]
```diff
@@ -56,12 +57,15 @@ ucp_memh_get(ucp_context_h context, void *address, size_t length,
                              uct_flags, memh_p);
 out_unlock:
     UCP_THREAD_CS_EXIT(&context->mt_lock);
+    ucs_trace("memh %p: get fast", *memh_p);
```
Comment (yosefe) [inline src/ucp/core/ucp_mm.inl:60]: "memh %p: address %p length %zu md_map %PRIx64/%PRIx64 obtained from rcache"

File: src/ucp/core/ucp_mm.inl [hunk 3]
```diff
@@ -56,12 +57,15 @@ ucp_memh_get(ucp_context_h context, void *address, size_t length,
...
     return status;
 }
 
 static UCS_F_ALWAYS_INLINE void
 ucp_memh_put(ucp_context_h context, ucp_mem_h memh, int invalidate)
 {
+    ucs_trace("memh %p: put", memh);
```
Comment (yosefe) [inline src/ucp/core/ucp_mm.inl:67]: cs_trace("memh %p: release address %p length %zu md_Map %PRIx64 invalidate: %d", ... memh);

File: src/ucp/core/ucp_mm.c (hunk 1 above)
Comment (evgeny-leksikov) [reply src/ucp/core/ucp_mm.c:434]: ok, initially I added only what was really useful in my case, I was seeking a memory leak.

File: src/ucp/core/ucp_mm.inl [hunk 4]
```diff
@@ -43,6 +44,9 @@ ucp_memh_get(ucp_context_h context, void *address, size_t length,
         memh = ucs_derived_of(rregion, ucp_mem_t);
 
         if (ucs_likely(ucs_test_all_flags(memh->md_map, reg_md_map))) {
+            ucs_trace("memh %p: address %p length %zu md_map %" PRIx64
+                      "/%" PRIx64 " obtained from rcache", memh, address,
+                      length, memh->md_map, reg_md_map);
```
Comment (yosefe) [inline src/ucp/core/ucp_mm.inl:49]: maybe swap order memh->md_map, reg_md_map

**Review by yosefe — DISMISSED** [review]

File: src/ucp/core/ucp_mm.inl [hunk 5]
```diff
@@ -43,6 +44,9 @@ ucp_memh_get(ucp_context_h context, void *address, size_t length,
         memh = ucs_derived_of(rregion, ucp_mem_t);
 
         if (ucs_likely(ucs_test_all_flags(memh->md_map, reg_md_map))) {
+            ucs_trace("memh %p: address %p length %zu md_map %" PRIx64
```
Comment (Artemy-Mellanox) [inline src/ucp/core/ucp_mm.inl:47]: * let's print in addition to address and length ucp_memh_address(memh)/length(memh) 
* let's add same print in [ucp_memh_get_slow()](https://github.com/openucx/ucx/blob/master/src/ucp/core/ucp_mm.c#L433)

File: src/ucp/core/ucp_mm.inl [hunk 6]
```diff
@@ -29,6 +29,7 @@ ucp_memh_get(ucp_context_h context, void *address, size_t length,
 
     if (length == 0) {
         *memh_p = &ucp_mem_dummy_handle.memh;
+        ucs_trace("memh %p: get dummy", *memh_p);
```
[the remark this answers is not in the document]
Comment (evgeny-leksikov) [reply src/ucp/core/ucp_mm.inl:35]: added assertions instead since it always should be `NULL` / `0`

[the remark this answers is not in the document]
Comment (Artemy-Mellanox) [reply src/ucp/core/ucp_mm.inl:35]: let's print original address, it shouldn't be NULL

**Review by Artemy-Mellanox — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Pull request merged**